src/Agda2Hs/Compile/Function.hs

```diff
@@ -82,58 +103,30 @@ compileLitNatPat = \case
     , [p] <- ps -> (1+) <$> compileLitNatPat (namedArg p)
   p -> genericDocError =<< "not a literal natural number pattern:" <?> prettyTCM p
 
-posIntPat :: ConHead -> ConPatternInfo -> [NamedArg DeBruijnPattern] -> C (Hs.Pat ())
-posIntPat c i [p] = do
-  n <- compileLitNatPat (namedArg p)
-  return $ Hs.PLit () (Hs.Signless ()) (Hs.Int () n (show n))
-posIntPat _ _ _ = __IMPOSSIBLE__
 
-negSucIntPat :: ConHead -> ConPatternInfo -> [NamedArg DeBruijnPattern] -> C (Hs.Pat ())
-negSucIntPat c i [p] = do
-  n <- (1+) <$> compileLitNatPat (namedArg p)
-  return $ Hs.PLit () (Hs.Negative ()) (Hs.Int () n (show (negate n)))
-negSucIntPat _ _ _ = __IMPOSSIBLE__
-
--- The bool argument says whether we also want the type signature or just the body
-compileFun, compileFun' :: Bool -> Definition -> C [Hs.Decl ()]
--- initialize locals when first stepping into a function
-compileFun withSig def@Defn{..} = setCurrentRangeQ defName $ do
-  -- check if the function is an operator with custom fixity,
-  -- then compile the function
-  maybePrependFixity $ withFunctionLocals defName $ compileFun' withSig def
-  where
-    Function{..} = theDef
-    m = qnameModule defName
-    n = qnameName defName
-    f@(Fixity _ lvl assoc) = n ^. lensFixity
-    x = hsName $ prettyShow n
+compileFun, compileFun'
+  :: Bool -- ^ Whether the type signature shuuld also be generated
+  -> Definition -> C [Hs.Decl ()]
+
+compileFun withSig def@Defn{..} =
+  setCurrentRangeQ defName 
+    $ maybePrependFixity defName (defName ^. lensFixity)
+    -- initialize locals when first stepping into a function
+    $ withFunctionLocals defName
+    $ compileFun' withSig def
 
-    checkFixityLevel :: FixityLevel -> C (Maybe Int)
-    checkFixityLevel Unrelated = pure Nothing
-    checkFixityLevel (Related lvl) =
-      if lvl /= fromInteger (round lvl) || lvl < 0
-        then genericDocError =<< text "Invalid fixity" <+> pretty lvl
-                             <+> text "for operator" <+> prettyTCM defName
-        else pure (Just (round lvl))
-
-    maybePrependFixity comp | f /= noFixity = do
-      hsLvl <- checkFixityLevel lvl
-      let hsAssoc = case assoc of
-            NonAssoc   -> Hs.AssocNone ()
-            LeftAssoc  -> Hs.AssocLeft ()
-            RightAssoc -> Hs.AssocRight ()
-      (Hs.InfixDecl () hsAssoc hsLvl [Hs.VarOp () x]:) <$> comp
-    maybePrependFixity comp = comp
--- inherit existing (instantiated) locals
```

Comment:
You dropped this comment, which might be helpful to someone coming across the `where`-reconstruction code for the first time.